cli/src/main.rs

```diff
@@ -240,6 +241,14 @@ async fn run_upload(
 
     let envs = EnvScanner::scan_env();
     let os_info: String = env::consts::OS.to_string();
+
+    let upload_op = Retry::spawn(default_delay(), || {
+        create_bundle_upload_intent(&api_address, &token, &org_url_slug, &repo.repo)
+    })
+    .await?;
+
+    let upload = upload_op.context("Failed to create bundle upload.")?;
```

Comment:
Was looking for a way to set `upload` here to a `CreateBundleUploadResponse` from an `Option<CreateBundleUploadResponse>`, failing otherwise, since I don't think we'd want to continue if `upload_op` were `None`. LMK if that's an incorrect assumption. Based on this SO thread :) https://stackoverflow.com/questions/69738600/simplest-way-to-unwrap-an-option-and-return-error-if-none-anyhow

Reply:
This is good, but I think we can do one better :)

In `create_bundle_upload_intent` I think we can and should change `Option<CreateBundleUploadResponse>` to just `CreateBundleUploadResponse` and then you won't need to do that here. That will also require changing from checking for `is_success` to checking if the API call is not successful (look at `get_quarantining_config` for example). This also makes `create_bundle_upload_intent` more consistent with our other API calls like `get_quarantining_config`. Eventually, I'll get around to cleaning this up so there's a better, more consistent pattern to all of the API calls with retries we make.

Removing the `Option<T>` type is a little auto-magical because it's how `serde` implicitly figures out how to deserialize the response using auto-implemented traits:
https://docs.rs/serde/latest/serde/de/index.html#implementations-of-deserialize-provided-by-serde

Reply:
Updated! lmk how it looks